tree/dataframe/inc/ROOT/RCsvDS.hxx

```diff
@@ -47,10 +47,12 @@ private:
    std::unique_ptr<ROOT::Internal::RRawFile> fCsvFile;
    const char fDelimiter;
    const Long64_t fLinesChunkSize;
+   const std::string fFillValue = "";
```

Comment:
If I understand correctly, this data member has 2 jobs that are in conflict:
- if a cell is empty, `fFillValue` is used instead
- if a cell contains `fFillValue`, its value is substituted to `0`, `false` or `nan` depending on the column type

So if I set a `fFillValue` of `-999`, because if a value is missing I want it to be `-999` instead (which is something that actually happens in HEP), then if a column value is actually `-999` I get `0` instead?

Reply:
`fFillValue` is now not exposed to the user. As discussed, this PR tries to handle the issue of reading NaNs, but not giving the freedom to change the fill value. The fill value is now always `""`, but it is easier to keep track of where it is needed.

Reply:
I see the concern though. There is a minor change that is now needed to ensure that we can handle user-defined `fFillValue` - in particular [here](https://github.com/ikabadzhov/root/blob/e2fe69f44395cd12fb2e384ba6bac6cddb9f7201/tree/dataframe/src/RCsvDS.cxx#L131) - in stead write `(col != fFillValue && fFillValue == "")` - and on few other places inside `FillRecord`.

This would be extra work done now, but would allow trivial extension to user-defined fill values. Should I add it?

Reply:
> this PR tries to handle the issue of reading NaNs

Right! We don't need `fFillValue` at all for that: if a cell value is `""` or `nan`, we need to transform that into 0/false/NaN. What am I missing?

Reply:
This was aimed at readability. I am removing the extra variable now.